google/cloud/spanner_v1/snapshot.py

```diff
@@ -149,8 +169,15 @@ def _restart_on_unavailable(
                 request.resume_token = resume_token
                 if transaction is not None:
                     transaction_selector = transaction._make_txn_selector()
+                nth_request = next_nth_request()
+                attempt = 1
```

Comment:
Same here: This is a retry of the query, and should use the same nth request and a higher attempt number.

Reply:
The very interesting thing is that I have been treating InternalServerError as a non-idempotent error: as it could result in inconsistent state as an operation might have succeeded on the server but it crapped out. However, good points, this makes it consistent with the other implementations and thank you.